### ISSUE TITLE
Double clicking opens duplicate notebook instances on MacOS

### ISSUE THREAD

**Issue by karlglazebrook** [jupyterlab/jupyterlab-desktop#586]
I was trying JupyterLab Desktop for the first time and set it to launch when double clicking on .ipynb files (which is nice).

The problem is it opens these in a new window each time AND duplicates all notebooks from the previous window!

So if I have files `Untitled1.ipynb Untitled2.ipynb Untitled3.ipynb Untitled4.ipynb `... etc

Then double clicking on `Untitled1.ipynb` gives me a window with a `Untitled1.ipynb` tab

If I then double click on `Untitled2.ipynb`  I get two windows -the new one has  `Untitled1.ipynb `AND  `Untitled2.ipynb` tabs while the old one still has the `Untitled1.ipynb` tab. Further they seem to be independent instances so editing one `Untitled1.ipynb `does not change the other. Lots of possibilities for conflicts.

If I double click on a third file this gets worse...I could not find a settings option to change this.

This really needs to be fixed!

(Version 3.6.1-1 on MacOS)

**Comment by mbektas** [comment]
@karlglazebrook thanks for reporting this, currently a new session is started for each file opened using double click. I plan to add options to open in the existing session when applicable.
for your use case, once you have JupyterLab Desktop open, you can go to file explorer on the side and open those other files in the same directory more easily and they will be sharing the same session.

**Comment by karlglazebrook** [comment]
Indeed, thanks for the response and sorry for the slow reply. I thought I had enabled notifications but clearly I had not.

'open in existing session' would be a valuable enhancement and provide a more integrated (in to the host OS) experience, which I think is part of the point of this app. It would also alleviate the fact that app startup is quite slow.

Meanwhile I will be careful and bare in mind your workaround.

Keep up the good work! This app is really needed.

**Comment by karlglazebrook** [comment]
PS I see my issue survived the triage. Great!

**Comment by karlglazebrook** [comment]
Any news on this? It still happening, if there is a setting I can't find it

**Comment by mbektas** [comment]
@karlglazebrook we haven't added support for this yet. sorry. I will keep this issue open since we would like to have this supported.

**Comment by karlglazebrook** [comment]
Yes please, it does drive me batty! :-)

**Comment by karlglazebrook** [comment]
There seems to be no progress on this? I thought it was a minor issue and seems relevant to new features like 'Zen mode' when opening files in MacOS finder.

**Comment by karlglazebrook** [comment]
A few more comments @mbektas 

- the issue where it DUPLICATES all the notebook tabs from the previous window, is actually I think a bug and unrelated to the enhancement request. It is not good behaviour to erroneously open two sessions pointing to the same notebook. This will cause accidental overwrites I think.

- further, this interacts in an unfortunate way with the zen mode. If I double click a single notebook in the Mac Finder and open it, it can open automatically in zen mode and then HIDE the other tabs (which potentially duplicate another session) which it opens in the bug case so you can't even see them.


Fixing the duplicates bug would be big win :-)

**Comment by karlglazebrook** [comment]
I was wondering if there is any progress on this issue? My naive understanding is that JLD stores the session info somewhere, after all that is why each instance reopens all the old notebooks, so it should be straight forward to connect to a currently running session?
